extensions/amp-addthis/0.1/amp-addthis.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds an AddThis widget.
+ * The data-pub-id and data-widget-id can be found easily in the AddThis dashboard at addthis.com.
+ * Example:
+ * <code>
+ * <amp-addthis
+ *   width="320"
+ *   height="92"
+ *   layout="responsive"
+ *   data-pub-id="ra-59c2c366435ef478"
+ *   data-widget-id="0fyg">
+ * </amp-addthis>
+ * </code>
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {setStyle} from '../../../src/style';
+import {createElementWithAttributes, removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+import {ConfigManager} from './ConfigManager';
+import {CUSTOM_SHARE_KEYS, ORIGIN, ICON_SIZE, ALT_TEXT} from './constants';
+
+// This `configManager` will be shared by all AmpAddThis elements on a page, to prevent unnecessary
+// HTTP requests, get accurate analytics, etc.
+const configManager = new ConfigManager();
+
+export function getConfigManager() {
+  return configManager;
+}
+
+class AmpAddThis extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframeLoadPromise_ = null;
```

Comment:
You  don't need this. It is not used anywhere outside of `layoutCallback`

Reply:
Good catch; thanks.